pkg/plugin/status.go

```diff
@@ -0,0 +1,206 @@
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	framework "github.com/kubemove/kubemove/pkg/plugin/ddm/plugin"
+)
+
+const (
+	SnapshotSucceeded  = "SUCCESS"
+	SnapshotInProgress = "IN_PROGRESS"
+	SnapshotFailed     = "FAILED"
+
+	RecoveryTypeSnapshot = "SNAPSHOT"
+	RecoveryStageDone    = "DONE"
+)
+
+// SnapshotGetResponse is used to unmarshal the response body of SnapshotGetRequest
+type SnapshotGetResponse struct {
+	Snapshots []SnapshotStatus `json:"snapshots"`
+}
+type SnapshotStatus struct {
+	Snapshot string `json:"snapshot"`
+	State    string `json:"state"`
+}
+
+// IndexesRecoveryStatus is used to unmarshal the indexes recovery status from IndicesRecoveryRequest response
+type IndexesRecoveryStatus struct {
+	Shards []ShardRecoveryStatus `json:"shards"`
+}
+type ShardRecoveryStatus struct {
+	Type   string         `json:"type"`
+	Stage  string         `json:"stage"`
+	Source RecoverySource `json:"source"`
+}
+type RecoverySource struct {
+	Snapshot string `json:"snapshot,omitempty"`
+}
+
+func (d *ElasticsearchDDM) Status(params map[string]string) (int32, error) {
+	fmt.Println("\nHandling STATUS request..................")
+	// extract the plugin parameters from the respective MoveEngine
+	pluginParameters, mode, err := extractPluginParameters(d.DmClient, params)
+	if err != nil {
```

Comment:
lets log this error. It will be better if we log the error wherever we are passing it to server.

Reply:
done